ui/components/FormDialog.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <simple-dialog
+    v-model="value"
+    class="form-dialog"
+    :class="`form-dialog_${mode}`"
+    :actions="actions"
+    :back-disabled="isFirstTab"
+    :details-route="detailsRoute"
+    :height="height"
+    :is-valid="isTabValid"
+    :on-submit="submit"
+    :title="computedTitle"
+    :width="width"
+    @back="goToPreviousTab"
+    @next="goToNextTab"
+  >
+    <template #sidebar>
+      <vertical-tabs
+        v-model:active-tab-key="activeTabKey"
+        :tabs="tabs"
+        :linear="mode === 'wizard'"
+      />
+    </template>
+    <template #default>
+      <slot :name="activeTabKey" />
+    </template>
+  </simple-dialog>
+</template>
+
+<script setup>
+import { ref, computed, watch } from 'vue';
+
+import SimpleDialog from '~/components/SimpleDialog.vue';
+import VerticalTabs from '~/components/VerticalTabs.vue';
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  tabs: {
+    type: Array,
+    required: true,
+  },
+  mode: {
+    type: String,
+    required: true,
+    validator(value) {
+      return ['wizard', 'edit'].includes(value);
+    },
+  },
+  form: {
+    type: Object,
+    required: true,
+  },
+  height: {
+    type: String,
+    default: '600px',
+  },
+  width: {
+    type: String,
+    default: '800px',
+  },
+  onSubmit: {
+    type: Function,
+    default: () => {},
+  },
+  detailsRoute: {
+    type: Object,
+    default: null,
+  },
+  rules: {
+    type: Object,
+    default: () => {},
+  },
+  submitLabel: {
+    type: String,
+    default: '',
+  },
+});
+
+const value = defineModel({
+  type: Boolean,
+  default: false,
+});
+
+const activeTabKey = ref('');
+const activeTabIdx = computed(() => props.tabs.findIndex((tab) => tab.key === activeTabKey.value));
+const activeTab = computed(() => props.tabs.find((tab) => tab.key === activeTabKey.value));
+const isFirstTab = computed(() => activeTabIdx.value === 0);
+const isLastTab = computed(() => activeTabIdx.value === props.tabs.length - 1);
```

Comment:
if the array is empty, this would be -1. does it explode?

Reply:
It would, yes, and this would break a lot of the component itself, since it meant to be used with tabs (its the only job it has basically).
Maybe adding a validator to the `tabs` prop, to force it to have items, would be better.
Something like:

```js
defineProps({
  ...
  tabs: {
    type: Array,
    required: true,
    validator(value) {
      return Boolean(value.length);
    },
  },
  ...
```

Reply:
Fixed, added the validator